src/middlewared/middlewared/scripts/zpoolhelper.py

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/python3
+from collections import deque
+from time import time, sleep
+from sys import exit
+
+from pyudev import Context, Monitor, MonitorObserver
+
+dq = deque()
+
+
+def callback(dev):
+    t = time()
+    if uuid := dev.get('ID_PART_ENTRY_UUID'):
+        dq.append({'time': t, 'name': dev.sys_name, 'uuid': uuid, 'action': dev.action})
+
+
+def get_observer():
+    ctx = Context()
+    mon = Monitor.from_netlink(ctx)
+    mon.filter_by('block')
+    return MonitorObserver(mon, callback=callback)
+
+
+def main():
+    obs = get_observer(dq)
+    obs.start()  # start background thread
+
+    max_time_to_wait = 600  # total time to wait in seconds
+    interval = 5  # seconds to sleep before checking for new event
+    last_event = {}
+    while max_time_to_wait > 0:
+        max_time_to_wait -= interval
+        sleep(interval)
+
+        try:
+            event = dq[-1]
```

Comment:
I am not sure what the size can be here but is there a good reason to use `deque` or `list` here as we are not really interested in keeping all the items ?

Reply:
`deque` is vastly more performant, vastly more memory efficient and the `.append` method is thread-safe 😄. The good news is that once `ix-zfs.service` exits memory is freed since this is a forked child process done by systemd.

Reply:
Nope, i meant that do we even need to use `list` or `deque` ?

Reply:
I think let's set `maxlen` for it to 5 or 10 ?

Reply:
Actually woudl require testing as iirc it would be blocking and won't require further additions

Reply:
I'd rather not set a length on the deque. The memory is freed once the script exits. There is no reason to artificially limit the size since it doesn't gain us anything.